tensorflow/python/ops/numpy_ops/np_array_ops.py

```diff
@@ -1066,6 +1066,10 @@ def stack(arrays, axis=0):  # pylint: disable=missing-function-docstring
     arrays = asarray(arrays)
     if axis == 0:
       return arrays
+    elif axis < -array_ops.rank(arrays) or axis >= array_ops.rank(arrays):
```

Comment:
The issue is actually in `swapaxes`, which doesn't properly check that the axis is in the valid range.

Reply:
@cantonios 

I have tested implementing range validation in swapaxes and it seems working fine. Attached [gist](https://colab.research.google.com/gist/SuryanarayanaY/81f511dd5eb0ecb4adf82e3dfd1260e1/60727_pr-r1.ipynb) for reference.

Validation Code and location can be found below.

```
def swapaxes(a, axis1, axis2):  # pylint: disable=missing-docstring
  a = asarray(a) #
  def adjust_axes(axes, rank):
    def f(x):
      if isinstance(x, int):
        if x < 0:
          x = x + rank
      else:
        x = array_ops.where_v2(x < 0, np_utils.add(x, a_rank), x)
      return x
    return nest.map_structure(f, axes)

  if axis2 < -array_ops.rank(a) or axis2 >= array_ops.rank(a):
    raise ValueError(
        f"Argument `axis` = {axis2} not in range "
                      f"[{-array_ops.rank(a)}, {array_ops.rank(a)})")  
```

Is it good to go implementing the validation code in `swapaxes` instead of `stack` ?

Reply:
The check should be in `swapaxes`, which has the same issue.  It currently assumes that `axis` is always in the range `[-rank, rank)`.

Reply:
Still waiting for the check to be made in `swapaxes`